header-translator/src/main.rs

```diff
@@ -0,0 +1,311 @@
+    };
+
+    let tu = index
+        .parser(&Path::new(env!("CARGO_MANIFEST_DIR")).join("framework-includes.h"))
+        .detailed_preprocessing_record(true)
+        .incomplete(true)
+        .skip_function_bodies(true)
+        .keep_going(true)
+        // .single_file_parse(true)
+        .include_attributed_types(true)
+        .visit_implicit_attributes(true)
+        // .ignore_non_errors_from_included_files(true)
+        .retain_excluded_conditional_blocks(true)
+        .arguments(&[
+            "-x",
+            "objective-c",
+            target,
+            "-Wall",
+            "-Wextra",
+            "-fobjc-arc",
+            "-fobjc-arc-exceptions",
+            "-fobjc-abi-version=2", // 3??
+            // "-fparse-all-comments",
+            "-fapinotes",
+            version_min,
+            "-isysroot",
+            sdk.path.to_str().unwrap(),
+        ])
+        .parse()
+        .unwrap();
+
+    println!("status: initialized translation unit {:?}", sdk.platform);
+
+    dbg!(&tu);
+    dbg!(tu.get_target());
+    dbg!(tu.get_memory_usage());
+    dbg!(tu.get_diagnostics());
+
+    // let dbg_file = |file: File<'_>| {
+    //     dbg!(
+    //         &file,
+    //         file.get_module(),
+    //         file.get_skipped_ranges(),
+    //         file.is_include_guarded(),
+    //         // file.get_includes(),
+    //         // file.get_references(),
+    //     );
+    // };
+    //
+    // dbg_file(tu.get_file(&header).unwrap());
+    // dbg_file(tu.get_file(&dir.join("NSAccessibility.h")).unwrap());
+    // let cursor_file = tu.get_file(&dir.join("NSCursor.h")).unwrap();
+    // dbg_file(cursor_file);
+
+    let entity = tu.get_entity();
+
+    dbg!(&entity);
+    dbg!(entity.get_availability());
+
+    let framework_dir = sdk.path.join("System/Library/Frameworks");
+    entity.visit_children(|entity, _parent| {
+        if let Some(location) = entity.get_location() {
+            if let Some(file) = location.get_file_location().file {
+                let path = file.get_path();
+                if let Ok(path) = path.strip_prefix(&framework_dir) {
+                    let mut components = path.components();
+                    let library = components
+                        .next()
+                        .expect("components next")
+                        .as_os_str()
+                        .to_str()
+                        .expect("component to_str")
+                        .strip_suffix(".framework")
+                        .expect("framework fileending");
+
+                    let path = components.as_path();
+                    let name = path
+                        .file_stem()
+                        .expect("path file stem")
+                        .to_string_lossy()
+                        .to_owned();
+
+                    f(&library, &name, entity);
+                }
+            }
+        }
+        EntityVisitResult::Continue
+    });
+}
+
+fn output_files(
```

Comment:
Should you want to use this crate as a [build script](https://doc.rust-lang.org/cargo/reference/build-scripts.html), I think `output_files` (and other stuff) will need to go in the `lib.rs`